lib/Dialect/Comb/CombFolds.cpp

```diff
@@ -145,6 +145,10 @@ narrowOperationWidth(Operation *narrowingCandidate, ValueRange inputs,
   auto loc = narrowingCandidate->getLoc();
   size_t narrowedWidth = highestBitRequired - lowestBitRequired + 1;
   auto narrowedType = rewriter.getIntegerType(narrowedWidth);
+
+  // Insert the new narrowedOperation at a point where all narrowingCandidate's
+  // operands are available, and resides before all users.
+  rewriter.setInsertionPoint(narrowingCandidate);
```

Comment:
I am not sure if it matters, but should this reset the insertion point when `narrowOperationWidth` exits? 

I think this can be done with an `InsertionGuard`. 

@lattner: do you know if this is needed here?

Reply:
I think this is a-okay as the insertion point should be reset before any rewriting happens here: https://github.com/llvm/llvm-project/blob/main/mlir/lib/Rewrite/PatternApplicator.cpp#L188.